transfer/src/main/java/com/kin/ecosystem/transfer/AccountInfoManager.java

```diff
@@ -0,0 +1,86 @@
+package com.kin.ecosystem.transfer;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.v4.content.FileProvider;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+
+class AccountInfoManager {
+
+    interface InitCompletionListener {
+        void onSuccess();
+
+        void onError(String reason);
+    }
+
+    private static String FILE_NAME = "accountInfo.txt";
+    private static String FILE_PROVIDER_NAME = "KinTransferAccountInfoFileProvider";
+    private static String FILE_PROVIDER_DIR_NAME = "kintransfer_account_info";
+
+    private File file;
+
+    //need to be called first
+    void init(@NonNull Context context, @NonNull String publicAddress, @NonNull InitCompletionListener listener) {
+        if(publicAddress.isEmpty()){
+            listener.onError("public address cant be empty");
+            return;
+        }
+        String dirPath = context.getFilesDir().getAbsolutePath() + File.separator + FILE_PROVIDER_DIR_NAME;
+        File dir = new File(dirPath);
+        if (!dir.exists()) {
+            if (!dir.mkdir()) {
+                listener.onError("Cant create directory " + dirPath);
+                return;
+            }
+        }
+        String fileFullPath = dirPath + File.separator + FILE_NAME;
+        try {
+            file = new File(fileFullPath);
+            if (!file.exists()) {
+                if (!file.createNewFile()) {
+                    file = null;
+                    listener.onError("Cant create file " + fileFullPath);
+                    return;
+                }
+            }
+            FileWriter writer = new FileWriter(file);
+            writer.append(publicAddress);
+            writer.flush();
+            writer.close();
+        } catch (IOException e) {
+            file = null;
+            listener.onError("Cant write to file " + e.getMessage());
+            return;
+        }
+        listener.onSuccess();
+    }
+
+    //Can be called anytime (before init complete), when user decline transfer
+    void respondCancel(@NonNull Activity activity) {
+        Intent intent = new Intent();
+        activity.setResult(Activity.RESULT_CANCELED, intent);
```

Comment:
file should be deleted if the user cancels

Reply:
or only create the file if the user approves...

Reply:
better to create before user approve cause it will delay the user if he press ok 
and then he needs to wait 
and what happen if it fails to create after send ok
better note the user before pressing something
not good idea to delete it cause other app needs to read it later on